HIP/hip-1000.md

```diff
@@ -0,0 +1,111 @@
+---
+hip:1000
+title: HIP Unlimited token associations per Account
+author: Anirudh Ghanta <anirudh.ghanta@hedera.com>
+working-group: a list of the technical and business stakeholders' name(s) and/or username(s), or name(s) and email(s).
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft <Draft | Review | Last Call | Active | Inactive | Deferred | Rejected | Withdrawn | Accepted | Final | Replaced>
+created: 2022-02-17
+discussions-to: <a URL pointing to the official discussion thread>
+updated: <comma separated list of dates>
+requires: <HIP number(s)>
+replaces: <HIP number(s)>
+superseded-by: <HIP number(s)>
+---
+
+## Abstract
+
+Describes the removal of the limit [1000] on the number of tokens an account can be associated to and the refactor of 
+how we store this **unlimited** list of associated tokens on the Account state. 
+
+
+## Motivation
+
+The current Services API allows the user to only have upto 1000 tokens associated at a time.
+For users that need more associations :
+- one can create multiple accounts to hold 1000 token associations each.
+- user can create a smartContract and manage the mapping of multiple accounts with 1000 token associations each.
+
+and a few others which are mentioned [here](https://www.notion.so/hedera/Increase-1000-associations-account-limit-95e882c6954443c6bc6502ad2818d15b#e8d1f192f620414c88a5ae18dfcca69b)
+
+## Rationale
+
+Rather, we should remove the limit we put in place for the token associations per account.
+
+And as the array of tokenIds that are associated to an account on the Account state grow, hashing/re-hashing every time
+we modify this array becomes expensive. A refactor of how we handle these associated tokenIds is needed.
```

Comment:
I think each section should stand alone. By starting the sentence with "Rather", we are continuing from the above section. Lets rework the wording. I'm also not sure how Rationale differs from Motivation?